packages/botonic-core/src/hubtype-service.js

```diff
@@ -35,36 +40,34 @@ export class HubtypeService {
     this.onEvent = onEvent
     this.unsentInputs = unsentInputs
     this.server = server
-    if (user.id && (lastMessageId || lastMessageUpdateDate)) {
+    if (this.user.id && (lastMessageId || lastMessageUpdateDate)) {
+      // It's safe not awaiting Promise because:
+      // * it will never be called from AWS lambda
+      // * though init() is called again from postMesage, it does nothing if Pusher already created
       this.init()
     }
   }
 
-  resolveServerConfig() {
-    if (!this.server) {
-      return { activityTimeout: ACTIVITY_TIMEOUT, pongTimeout: PONG_TIMEOUT }
-    }
-    return {
-      activityTimeout: this.server.activityTimeout || ACTIVITY_TIMEOUT,
-      pongTimeout: this.server.pongTimeout || PONG_TIMEOUT,
-    }
-  }
-
-  updateAuthHeaders() {
-    if (this.pusher) {
-      this.pusher.config.auth.headers = {
-        ...this.pusher.config.auth.headers,
-        ...this.constructHeaders(),
-      }
-    }
-  }
-
+  /**
+   * @returns {Promise<void>}
+   */
   init(user, lastMessageId, lastMessageUpdateDate) {
     if (user) this.user = user
     if (lastMessageId) this.lastMessageId = lastMessageId
     if (lastMessageUpdateDate)
       this.lastMessageUpdateDate = lastMessageUpdateDate
```

Comment:
Question here: are you ensuring those properties are not overwritten in case they are passed `null` | `undefined`, so they keep the values set on the constructor if they were provided?

It looks weird passing mandatory values that may be set `null` on purpose and then it sends different ones.

Reply:
I agree it's not a great design, but I understand that the PR is not affecting this. In the future I'd also like to apply SRP and separate the Pusher part to its own class